cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/passes/SymbolResolver.kt

```diff
@@ -479,6 +480,85 @@ open class SymbolResolver(ctx: TranslationContext) : ComponentPass(ctx) {
             return
         }
 
+        // Let's do some next generation stuff here, but currently only for "normal" function calls
+        val nextGen =
+            when {
+                curClass != null -> false
+                call is MemberCallExpression -> false
+                call is ConstructExpression -> false
+                call.instantiatesTemplate() && call.language is HasTemplates -> false
+                call.callee !is Reference -> false
+                else -> {
+                    // Unfortunate we need to resort the old symbol resolving if templates are
+                    // involved and since we do not have the template expansion pass yet, we need to
+                    // a rather stupid check, if they are possible template functions with this name
+                    // and disable NG.
+                    val templateFunctions =
+                        scopeManager.resolve<FunctionTemplateDeclaration>(scope) {
+                            it.name.lastPartsMatch(call.name)
+                        }
+                    templateFunctions.isEmpty()
+                }
+            }
+
+        if (nextGen) {
+            handleCallExpression(call, curClass)
+        } else {
+            handleCallExpressionLegacy(call, curClass)
+        }
+    }
+
+    /**
+     * This function tries to resolve the call expression according to the [CallExpression.callee].
+     */
+    private fun handleCallExpression(call: CallExpression, curClass: RecordDeclaration?) {
+        val result = scopeManager.resolveCall(call)
+
+        when (result.type) {
+            ScopeManager.CallResolutionResult.Type.PROBLEMATIC -> {
+                log.error(
+                    "Resolution of ${call.name} returned an problematic result and we cannot decide correctly, the invokes edge will contain all possible candidates"
+                )
+                call.invokes = result.bestViable.toList()
+            }
+            ScopeManager.CallResolutionResult.Type.AMBIGUOUS -> {
+                log.error(
```

Comment:
Is `log.error` appropriate here or is it only a warning?

Reply:
A warning is probably better.

Reply:
A would keep an error for "PROBLEMATIC" though